network/src/protocols/wire/messaging/v1/mod.rs

```diff
@@ -2,41 +2,53 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! This module defines the structs transported during the network messaging protocol v1.
-//! These should serialize as per [link](TODO: Add ref).
+//! These should serialize as per the [specification](https://github.com/libra/libra/blob/master/specifications/network/messaging-v1.md)
 
-use crate::protocols::wire::handshake::v1::{MessagingProtocolVersion, ProtocolId};
+use crate::protocols::wire::handshake::v1::ProtocolId;
 use serde::{Deserialize, Serialize};
 
 #[cfg(test)]
 mod test;
 
-/// Message variants that are sent on the wire.
-/// New variants cannot be added without bumping up the MessagingProtocolVersion.
+/// Most primitive message type set on the network.
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub enum NetworkMessage {
     Error(ErrorCode),
-    Ping(Nonce),
-    Pong(Nonce),
     RpcRequest(RpcRequest),
     RpcResponse(RpcResponse),
     DirectSendMsg(DirectSendMsg),
 }
 
-/// Enum representing various error codes that can be embedded in NetworkMessage.
-/// New variants cannot be added without bumping up the MessagingProtocolVersion.
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub enum ErrorCode {
-    /// Ping timed out.
-    TimedOut,
     /// Failed to parse NetworkMessage when interpreting according to provided protocol version.
-    ParsingError(MessagingProtocolVersion, Box<NetworkMessage>),
+    ParsingError(ParsingErrorType),
     /// A message was received for a protocol that is not supported over this connection.
-    NotSupported(ProtocolId),
+    NotSupported(NotSupportedType),
 }
 
-/// Nonces used by Ping and Pong message types.
+impl ErrorCode {
+    pub fn parsing_error(message: u8, protocol: u8) -> Self {
```

Comment:
use ProtocolId instead of u8?

Reply:
for the parsing error, the message and protocol type may not be within our enum... consider this the "safe" version of the stricter unsupported protocol.